MobileBuy/buy/src/main/java/com/shopify/buy3/GraphClient.java

```diff
@@ -0,0 +1,137 @@
+
+  private OkHttpClient buildHttpCallFactory(final String apiKey, final String applicationName, final long httpConnectionTimeoutMs,
+                                            final long httpReadWriteTimeoutMs, final Interceptor[] httpInterceptors) {
+    final Interceptor requestInterceptor = chain -> {
+      final Request original = chain.request();
+      final Request.Builder builder = original.newBuilder().method(original.method(), original.body());
+      builder.header("Authorization", Utils.formatBasicAuthorization(apiKey));
+
+      // Using the full package name for BuildConfig here as a work around for Javadoc.
+      builder.header("User-Agent", "Mobile Buy SDK Android/" + BuildConfig.VERSION_NAME + "/" + applicationName);
+
+      return chain.proceed(builder.build());
+    };
+
+    final OkHttpClient.Builder builder = new OkHttpClient.Builder()
+      .connectTimeout(httpConnectionTimeoutMs, TimeUnit.MILLISECONDS)
+      .readTimeout(httpReadWriteTimeoutMs, TimeUnit.MILLISECONDS)
+      .writeTimeout(httpReadWriteTimeoutMs, TimeUnit.MILLISECONDS)
+      .addInterceptor(requestInterceptor);
+
+    if (httpInterceptors != null) {
+      for (Interceptor interceptor : httpInterceptors) {
+        builder.addInterceptor(interceptor);
+      }
+    }
+
+    return builder.build();
+  }
+
+  public static final class Builder {
+    public static final long DEFAULT_HTTP_CONNECTION_TIME_OUT_MS = TimeUnit.SECONDS.toMillis(30);
+    public static final long DEFAULT_HTTP_READ_WRITE_TIME_OUT_MS = TimeUnit.SECONDS.toMillis(60);
+
+    private String shopDomain;
+    private String apiKey;
+    private String applicationName;
+    private Interceptor[] httpInterceptors;
+    private long httpConnectionTimeoutMs = DEFAULT_HTTP_CONNECTION_TIME_OUT_MS;
+    private long httpReadWriteTimeoutMs = DEFAULT_HTTP_READ_WRITE_TIME_OUT_MS;
+
+    /**
+     * Sets store domain url (usually {store name}.myshopify.com
+     *
+     * @param shopDomain The domain for the shop.
+     * @return a {@link BuyClientBuilder}
+     */
+    public Builder shopDomain(final String shopDomain) {
+      this.shopDomain = shopDomain;
+      return this;
+    }
+
+    /**
+     * Sets Shopify store api key
+     *
+     * @param apiKey The Api Key.
+     * @return a {@link BuyClientBuilder}
+     */
+    public Builder apiKey(final String apiKey) {
+      this.apiKey = apiKey;
+      return this;
+    }
+
+    /**
+     * Sets Shopify store application name
+     *
+     * @param applicationName The application name.
+     * @return a {@link BuyClientBuilder}
+     */
+    public Builder applicationName(final String applicationName) {
+      this.applicationName = applicationName;
+      return this;
+    }
+
+    /**
+     * Sets custom OkHttp httpInterceptors
+     *
+     * @param httpInterceptors Interceptors to add to the OkHttp client.
+     * @return a {@link BuyClientBuilder}
+     */
+    public Builder httpInterceptors(final Interceptor... httpInterceptors) {
```

Comment:
hmm, I guess we could still hook into these to do a simple http cache even with this client